### ISSUE TITLE
FT-4 — MDP mode: dipole/polarizability fine-tuning of the dielectric family

### ISSUE THREAD

**Issue by aacostadiaz** [ACEsuit/mace#1637]
**Depends on:** FT-1 (#1580), FT-3 (#1582), CFG-1 (#1574), TRN-2 (#1576), DEP-1a (#1634), ELEC-2 (#1592) · **Blocks:** GATE-3 (#1585)

**Prerequisite:** the Phase 0 characterization suite is committed — goldens, fixtures and the single-source tolerance table under `tests/golden/`, characterization tests under `tests/unit/`. References of the form `P0-N` name that work.

**Context:** `--finetune_dipoles_polarizabilities` (`mace/tools/arg_parser.py:1058-1060`) is a whole fine-tuning **mode** that no other ticket covers — v1 without it is a regression. It fine-tunes a pretrained `AtomicDielectricMACE` (MACE-MDP) on dipoles and polarizabilities only, and legacy implements it as a block of interacting namespace rewrites at the top of `run()` (`mace/cli/run_train.py:133-162`): it asserts `--model AtomicDielectricMACE` and a `--foundation_model`, rejects any `--loss` other than `weighted`/`dipole_polar`, then sets `args.loss = "dipole_polar"`, disables `args.multiheads_finetuning` (because the multihead default would otherwise force `loss="universal"` at `:435`), and downgrades `--E0s=foundation|estimated` to `average` with a warning because `AtomicDielectricMACE` has no `atomic_energies_fn`. A second check at `:220-227` verifies the loaded foundation checkpoint really is an `AtomicDielectricMACE`. Weight transfer runs through its own function, `load_foundations_mdp` (`mace/tools/finetuning_utils.py:711`), which repeats the `species_scale = (num_species_foundations / num_species) ** 0.5` subset slicing but additionally transfers higher-order `skip_tp` irreps and every angular-momentum channel of the products — behaviour `load_foundations_elements_default` does not have. The deployed side already exists too: `mace_mdp` downloads and wraps the published MACE-MDP artifact (`mace/calculators/foundations_models.py:141-142,:636`) as a `MACECalculator(model_type="DipolePolarizabilityMACE")`. Coverage is two workflow tests (`tests/workflows/test_mdp_finetune.py`: a train-then-finetune-then-params-changed case and a wrong-model-type rejection) plus three unit tests (`tests/unit/test_calculator_mdp.py`). This ticket re-lands the mode as **config**, on FT-1 (#1580)'s head interface and TRN-2 (#1576)'s generated losses. It depends on ELEC-2 (#1592) because that is where the dielectric model itself arrives on the v1 architecture; FT-4 (#1637) owns the fine-tune mode, not the model.

**Interface & constraints:**

1. **The mode is a config shape, not a preamble of mutations.** `finetune.mode = "dielectric"` (or, better, the absence of a mode: a config that declares only `dipole` and `polarizability` observables and a dielectric foundation model). Every legacy rewrite becomes either a declared default or a validator:
   - the `dipole_polar` loss is what declaring `dipole` + `polarizability` observables generates (TRN-2 (#1576)), not a string assignment;
   - "multihead finetuning off" is the resolved default for a dielectric foundation model, visible in the resolved config;
   - an `e0s` field on a dielectric model config is a **validation error** (CFG-1 (#1574)'s dipole-forbids-e0s rule), replacing the warn-and-downgrade-to-`average` at `run_train.py:153-159`;
   - a `--foundation_model` whose artifact metadata does not declare the dielectric family is a validation error at config-resolution time, not a class-name check after the checkpoint is unpickled (`run_train.py:220-227`).
2. **The transfer path is one implementation with declared options, not a third function.** `load_foundations_mdp`'s extra behaviour — higher-order `skip_tp` irreps and all angular-momentum channels of the products — is expressed as options on FT-3 (#1582)'s single element-transfer operation over canonical tensors, so the dielectric family stops needing its own copy of the slicing arithmetic. The `species_scale` rescale applies only on the explicit subset path, exactly as in FT-3 (#1582).
3. **Trainable surface is explicit.** Legacy's mode is "naive: all params trainable" (documented in `tests/workflows/test_mdp_finetune.py:112`). v1 keeps that as the default but expresses it through FT-1 (#1580)'s freezing/LoRA fields, so a user can restrict it without a new flag.
4. **The deployed path stays reachable.** `mace_mdp` (`foundations_models.py:611`) becomes an FM-3 (#1588) registry entry; the calculator side (`model_type="DipolePolarizabilityMACE"`, `get_dielectric_derivatives`) is DEP-1a (#1634). FT-4 (#1637) only guarantees that a model fine-tuned through this mode loads into it.

**Task:**
1. Add the dielectric fine-tune shape to the `finetune` config section: declared defaults (multihead off) and the three validators (forbidden `e0s`, family check against artifact metadata, loss/observable coherence).
2. Express `load_foundations_mdp`'s extra transfer coverage as options on FT-3 (#1582)'s element-transfer operation; delete the need for a family-specific transfer function in v1.
3. Wire the mode through FT-1 (#1580)'s head interface and TRN-2 (#1576)'s generated `dipole` + `polarizability` loss terms.
4. Port both `tests/workflows/test_mdp_finetune.py` cases under `--engine v1`, plus a parity case against legacy `load_foundations_mdp` on a tiny dielectric anchor at fp64.
5. Assert a model fine-tuned through this mode loads into the DEP-1a (#1634) calculator and reproduces its dipoles/polarizabilities.

**Out of scope:** the dielectric model itself on the v1 architecture (ELEC-2 (#1592)); the calculator surface and `get_dielectric_derivatives` (DEP-1a (#1634)); the `mace_mdp` registry entry and its download (FM-3 (#1588)); the polar density-cube CLI (CLI-1 (#1579) slot, ELEC-2 (#1592) implementation); element-table extension (FT-3 (#1582)).

**Acceptance criteria:**
- [ ] A dielectric fine-tune runs from a v1 config with no mode-specific flag: declaring `dipole` + `polarizability` observables and a dielectric foundation model is sufficient, and the resolved config shows multihead finetuning off as a declared default.
- [ ] An `e0s` field on a dielectric config, a non-dielectric foundation artifact, and an incompatible declared loss each fail validation with a message naming the field — none of them warns-and-continues.
- [ ] The single element-transfer operation, with the dielectric options set, reproduces legacy `load_foundations_mdp` output at fp64 on a tiny anchor (including the `species_scale` division on the subset path).
- [ ] Both ported `test_mdp_finetune.py` cases pass under `--engine v1`; parameters demonstrably change and the wrong-model-type case exits nonzero.
- [ ] The fine-tuned artifact loads into the DEP-1a (#1634) calculator and reproduces its dipoles and polarizabilities at fp64.

**Inventory gaps assigned here:**
- [ ] `--finetune_dipoles_polarizabilities` — two completion-level workflow tests today; nothing pins the transferred weights or the four rewrites the mode performs.

**Verify:**
```bash
MACE_ENGINE=v1 python -m pytest tests/workflows/test_mdp_finetune.py -v
python -m pytest tests/unit/test_calculator_mdp.py -v
python -m pytest tests/parity -m "not slow" -k mdp        # vs legacy load_foundations_mdp
```
**Review focus:** that the four legacy namespace rewrites all became either a declared default or a validator (none of them survived as a runtime mutation), and the transfer-option parity — the dielectric path touches tensor families the default path does not, so a missed channel is a silent accuracy loss.

---
